spieleplattform-backend/src/main/kotlin/com/example/spieleplattformbackend/gameConsole/GameConsoleService.kt

```diff
@@ -1,10 +1,20 @@
 package com.example.spieleplattformbackend.gameConsole
 
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 
 @Service
 class GameConsoleService(val gameConsoleRepository: GameConsoleRepository) {
     fun getGameConsoles(): Iterable<GameConsole> {
         return gameConsoleRepository.findAll()
     }
+
+    fun getGameConsolesByGameConsoleIds(gameConsoleIds: MutableList<Int>): MutableList<GameConsole>? {
+        val gameConsoleList = mutableListOf<GameConsole>()
+        for (id in gameConsoleIds) {
```

Comment:
Auch hier könntest du mit `map()` um den for-loop herumkommen. Die Funktion hat allerdings noch ein anderes Problem: Für jede Console die du hier abrufst, baust du eine neue Verbindung zu deiner Datenbank auf und Datenbankaufrufe sind immer recht teuer, was die Performance angeht. Besser würdest du dir also eine Repository function schreiben, die das hier für dich übernimmt. Ich glaube es gibt auch eine `findAllByIdOrNull` Methode Standardmäßig auf dem Crud repo. Wenn nicht, müsstest du dir halt selbst ein query schreiben.